### PR TITLE
perf(app): prevent superfluous top level re-renders

### PR DESCRIPTION
**Pull request by korhaliv**
## Description:
Prevent unnecessary `App` re-renders by extracting peer poll timeout from the `state` into a ref

## Motivation and Context:
`nextFetchIn` doesn't  influence anything in render func so can be safely extracted into a ref

## How Has This Been Tested?
Manually by monitoring redux log

## Types of changes:
Refactor/perf improvement

## Checklist:

- [x] My code follows the code style of this project.
- [x] I have reviewed and updated the documentation accordingly.
- [x] I have read the _CONTRIBUTING_ document.
- [ ] I have added tests to cover my changes where needed.
- [ ] All new and existing tests passed.
- [x] My commits have been squashed into a concise set of changes.

### CODE REVIEW

File: utils/scheduler.js [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+      return taskDesc
+    }
+
+    // else try to find by task callback
+    return Object.values(taskList).find(entry => entry.task === task)
+  }
+
+  /**
+   * Enqueues `task` for the continuous execution
+   * @param {Object} taskDefinition - wallet location desc
```
Comment (mrfelton) [inline utils/scheduler.js]: `wallet location desc`?

File: utils/scheduler.js [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+  const onCancelComplete = task => {
+    const taskDesc = findTask(task)
+    if (taskDesc) {
+      const keyToRemove = Object.keys(taskList).find(key => taskList[key] === taskDesc)
+      if (keyToRemove) {
+        delete taskList[keyToRemove]
+      }
+    }
+  }
+  /**
```
Comment (mrfelton) [inline utils/scheduler.js]: Add newline

File: utils/scheduler.js [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+    }
+  }
+}
+/**
+ * Creates scheduler instance. It is allowed to have unlimited scheduler instances
+ * Schedulers allow to add and remove tasks with flat or backoff schedules
+ */
+export default function createScheduler() {
+  const taskList = {}
+  /**
```
Comment (mrfelton) [inline utils/scheduler.js]: Add newline

File: utils/scheduler.js [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+        getTask,
+        baseDelay: maxDelay ? Math.min(maxDelay, nextDelay) : nextDelay,
+        maxDelay,
+        checkIsCancelled,
+        backoff,
+      })
+    }
+  }
+}
+/**
```
Comment (mrfelton) [inline utils/scheduler.js]: Add newline

File: utils/scheduler.js (hunk 1 above)
Comment (korhaliv) [reply utils/scheduler.js]: copy paste to rule the world

**Comment by korhaliv** [comment]
@mrfelton all updated

**Pull request merged**